www/thumbnail.php

```diff
@@ -0,0 +1,95 @@
+<?php
+if (file_exists(dirname(__FILE__).'/../config.inc.php')) {
+    require_once dirname(__FILE__).'/../config.inc.php';
+} else {
+    require dirname(__FILE__).'/../config.sample.php';
+}
+
+// Send headers for CORS support so poster images can be used when crossorigin="anonymous"
+header('Access-Control-Allow-Origin: *');
+header('Access-Control-Allow-Methods: GET, OPTIONS');
+header('Access-Control-Allow-Headers: X-Requested-With');
+if (isset($_SERVER['REQUEST_METHOD'])
+    && $_SERVER['REQUEST_METHOD'] == 'OPTIONS') {
+    // short circuit execution for CORS OPTIONS reqeusts
+    exit();
+}
+
+//Validate input
+if (!isset($_GET['media_id'])) {
+    throw new \Exception('You must provide a media_id', 400);
+}
+
+$media_id = (int)$_GET['media_id'];
+
+/**
+ * Create the cache key
+ * Note that the cache key does not contain the time code (only the url of the media).
+ * To create thumbnail at at specific time of the video, you must make a request to something like
+ * thumbnail.php?rebuild&time=00:00:10.00&url=x
+ * Subsequent requests to thumbnail.php?url=x will yield the last generated thumbnail
+ */
+$directory = UNL_MediaHub::getRootDir() . '/www/uploads/thumbnails/'.$media_id;
+$file = $directory.'/original.jpg';
+
+if (isset($_GET['rebuild']) || !file_exists($file)) {
```

Comment:
Instead of a giant `if` wrapper, it would look cleaner to check the opposite and exit (thus avoiding the else case).